docs/source/config_file.rst

```diff
@@ -744,25 +752,28 @@ section of the command line docs.
 
     Make arguments prepended via ``Concatenate`` be truly positional-only.
 
+    .. warning::
+       Deprecated: use :confval:`extra_checks` instead.
```

Comment:
I'm not sure about making this change while dsicussion in https://github.com/python/mypy/issues/18070 is ongoing